share/functions/fish_vi_key_bindings.fish

```diff
@@ -141,8 +177,34 @@ function fish_vi_key_bindings --description 'vi-like key bindings for fish'
     # Default (command) mode
     bind -s --preset :,q exit
     bind -s --preset -m insert ctrl-c clear-commandline repaint-mode
-    bind -s --preset -M default h backward-char
-    bind -s --preset -M default l forward-char
+
+    for i in (seq 1 9)
+        bind -s --preset -M default $i "fish_vi_arg_digit $i"
+    end
+    # 0 is special: it is 'beginning-of-line' unless we are already counting (e.g. 10)
+    bind -s --preset -M default 0 "if test -n \"\$__fish_vi_count\"; fish_vi_arg_digit 0; else; commandline -f beginning-of-line; end"
+
+    # --- Movement with Count Support ---
+    bind -s --preset -M default h 'fish_vi_run_count backward-char'
+    bind -s --preset -M default l 'fish_vi_run_count forward-char'
```

Comment:
this approach looks good but let's reset the count on escape, like Vim does.

In future, we can also add count support to arrow keys,
probably by passing fish_vi_run_count as argument to __fish_shared_key_bindings.